src/TaskEither.ts

```diff
@@ -165,6 +165,14 @@ export const fromIOEither = <L, A>(fa: IOEither<L, A>): TaskEither<L, A> => {
   return new TaskEither(taskFromIO(fa.value))
 }
 
+/**
+ * @function
+ * @since 1.6.0
+ */
+export const fromPredicate = <L, A>(predicate: Predicate<A>, whenFalse: (a: A) => L) => (a: A): TaskEither<L, A> => {
```

Comment:
This can be optimized

```ts
export const fromPredicate = <L, A>(predicate: Predicate<A>, whenFalse: (a: A) => L): ((a: A) => TaskEither<L, A>) => {
  const f = eitherFromPredicate(predicate, whenFalse)
  return a => fromEither(f(a))
}
```